src/utils/index.ts

```diff
@@ -1,10 +1,15 @@
 import axios from "axios";
 import Papa from "papaparse";
+import { db } from "../indexedDB";
 
 const root =
   "https://raw.githubusercontent.com/MoH-Malaysia/covid19-public/main/";
 
-export const fetcher = async <T extends {[key: string]: string}>(url: string): Promise<T[]> => {
+export const fetcher = async <T extends { [key: string]: string }>(
+  url: string
+): Promise<T[]> => {
+  const record = await db.source.get(1);
+  if ((record && record.url) === root + url) return await db.store.toArray();
```

Comment:
I think this will impact the performance of the application, because we have to wait for the response of the db every time we use the fetcher function, what do you think @Yuenmin ?

Reply:
Yes, it will be slightly affected, but I think it is acceptable as in the source table will have at most 1 record only.